src/sqlfluff/core/templaters/placeholder.py

```diff
@@ -144,6 +147,17 @@ def process(
             else:
                 param_name = found_param["param_name"]
             last_literal_length = span[0] - last_pos_raw
+
+            if (
+                context.get("autofill_missing_params")
+                and param_name not in context.keys()
+            ):
+                templater_logger.info(
+                    f"Autofilling missing parameter {param_name} with {AUTOFILL_PARAMS['string']}"
+                )
+                # TODO generate from types if exist
```

Comment:
This is a fairly big issue for this PR:

The current implementation works for the specific case that the we are substituting in an object (should be a string I guess but doesn't have quotes atm). 
![image](https://user-images.githubusercontent.com/80432516/145734717-b6420c25-1814-4f35-a11d-e7c76e0d8428.png)

But this is going to cause parsing errors whenever a string is not appropriate:
![image](https://user-images.githubusercontent.com/80432516/145734809-bd2e54ce-c2eb-4371-b63e-ec9d656a4d6d.png)
This is misleading to the user that there's something wrong with their query rather than that they've just forgotten to add a variable.

Or what if the intended substitution type is not clear or a mix or includes complicated structure and this substitution could cause the query to be interpreted differently or have a parsing error:
![image](https://user-images.githubusercontent.com/80432516/145735054-f442909a-a57c-4628-ac1f-2e64e1022205.png)
![image](https://user-images.githubusercontent.com/80432516/145735485-d86272db-84a6-49e1-941c-263cde66c4a1.png)

The big issue is templating happens before parsing, as how are you supposed to parse code without knowing what the code is, but then in your case if you haven't parsed the code then how can you interpret what type is implied?

That's the motivation for us having users provide their own values as a template could feasibly hold anything and only they'll 100% know what is intended.

Reply:
definitely can't be merged in as is given the potential parse issues.